src/main/java/com/commercetools/payment/handler/CommercetoolsExecutePaymentsController.java

```diff
@@ -26,13 +30,15 @@ public CommercetoolsExecutePaymentsController(@Nonnull StringTrimmerEditor strin
 
     @RequestMapping(
             method = RequestMethod.POST,
+            consumes = APPLICATION_JSON_VALUE,
+            produces = APPLICATION_JSON_VALUE,
             value = "/{tenantName}/commercetools/execute/payments/")
     public ResponseEntity executePayments(@PathVariable String tenantName,
-                                                 @RequestParam String paypalPlusPaymentId,
-                                                 @RequestParam String paypalPlusPayerId) {
+                                          @Valid @RequestBody PaypalPlusExecuteParams params,
+                                          BindingResult result) {
         PaymentHandleResponse paymentHandleResponse = paymentHandlerProvider
                 .getPaymentHandler(tenantName)
-                .map(paymentHandler -> paymentHandler.executePayment(paypalPlusPaymentId, paypalPlusPayerId))
+                .map(paymentHandler -> paymentHandler.executePayment(params.getPaypalPlusPaymentId(), params.getPaypalPlusPayerId()))
```

Comment:
Posted error response is not user friendly and exposes unrelated internals generated by spring framework. You could skip validation by Spring and validate by yourself or if possible configure Spring to return nice error messages only like that:

`Required field 'paypalPlusPayerId' is not set.` or `Required field 'paypalPlusPaymentId' is not set paypalPlusPaymentId`